db/import_listener.go

```diff
@@ -130,35 +130,43 @@ func (il *importListener) StartImportFeed(dbContext *DatabaseContext) (err error
 // internal documents based on key, then checks sync metadata to determine whether document needs to be imported
 func (il *importListener) ProcessFeedEvent(event sgbucket.FeedEvent) (shouldPersistCheckpoint bool) {
 
+	ctx := il.loggingCtx
 	// Ignore non-mutation/deletion events
 	if event.Opcode != sgbucket.FeedOpMutation && event.Opcode != sgbucket.FeedOpDeletion {
 		return true
 	}
-	key := string(event.Key)
+	docID := string(event.Key)
+
+	collection, ok := il.collections[event.CollectionID]
+	if !ok {
+		base.WarnfCtx(ctx, "Received import event for unrecognised collection 0x%x", event.CollectionID)
+		return
+	}
+	ctx = base.CollectionLogCtx(ctx, collection.Name)
 
 	// Ignore internal documents
-	if strings.HasPrefix(key, base.SyncDocPrefix) {
+	if strings.HasPrefix(docID, base.SyncDocPrefix) {
 		// Ignore all DCP checkpoints no matter config group ID
-		return !strings.HasPrefix(key, base.DCPCheckpointRootPrefix)
+		return !strings.HasPrefix(docID, base.DCPCheckpointRootPrefix)
 	}
 
 	// If this is a delete and there are no xattrs (no existing SG revision), we shouldn't import
 	if event.Opcode == sgbucket.FeedOpDeletion && len(event.Value) == 0 {
-		base.DebugfCtx(il.loggingCtx, base.KeyImport, "Ignoring delete mutation for %s - no existing Sync Gateway metadata.", base.UD(event.Key))
+		base.DebugfCtx(ctx, base.KeyImport, "Ignoring delete mutation for %s - no existing Sync Gateway metadata.", base.UD(docID))
 		return true
 	}
 
 	// If this is a binary document we can ignore, but update checkpoint to avoid reprocessing upon restart
 	if event.DataType == base.MemcachedDataTypeRaw {
-		base.InfofCtx(il.loggingCtx, base.KeyImport, "Ignoring binary mutation event for %s.", base.UD(event.Key))
+		base.InfofCtx(ctx, base.KeyImport, "Ignoring binary mutation event for %s.", base.UD(docID))
 		return true
 	}
 
-	il.ImportFeedEvent(event)
+	il.ImportFeedEvent(ctx, collection, event)
 	return true
 }
 
-func (il *importListener) ImportFeedEvent(event sgbucket.FeedEvent) {
+func (il *importListener) ImportFeedEvent(ctx context.Context, collection DatabaseCollectionWithUser, event sgbucket.FeedEvent) {
```

Comment:
As you've noted, I think this does need to be a pointer to allow `ReloadUser` to change the user inside this struct - I'm not sure whether this function is called from here or not.

Reply:
I think user doesn't matter because there's no user associated with import feed.

Reply:
Should this be a pointer in any case, to avoid any unwanted copying?

Reply:
Made a ptr.